### PR TITLE
Add support for discussion comment

### PR DESCRIPTION
**Pull request by khiga8**
Fixes: https://github.com/github/accessibility-alt-text-bot/issues/11

This PR adds support for discussion comments. Because the `action.yml` is getting big, I introduced a `queries.sh` file to move the GraphQL calls into, including the mutation call we introduced in https://github.com/github/accessibility-alt-text-bot/pull/14. 

We are using the [`addDiscussionComment`](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#adddiscussioncomment) and setting the `replyToId` to post a comment as a reply in a thread.

When replying to a top-level discussion comment, we're able to use `github.event.comment.node_id` as the `replyToId` to post a reply successfully.

However, I ran into issues making the same call when replying to a discussion comment inside of a thread. I kept running into the error:

```bash
gh: Parent comment is already in a thread, cannot reply to it
```

I discovered that `replyToId` has to be a top-level/parent comment node id. Since the `github.event.comment.node_id` only contains the node ID of the comment we're replying to, we must make a GraphQL query to get the parent comment node ID. We do that in the function `getDiscussionReplyToId`.

### CODE REVIEW

File: queries.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  '
+}
+
+# Given a discussion node ID, discussion comment node ID, and a message, adds a discussion comment as a reply in thread.
+function addDiscussionCommentAsReply() {
+  local DISCUSSION_NODE_ID=$1
+  local REPLY_TO_ID=$2
+  local MESSAGE=$3
+  gh api graphql -F discussionId="$DISCUSSION_NODE_ID" -F replyToId="$REPLY_TO_ID" -F body="$MESSAGE" -f query='
```
[the remark this answers is not in the document]
Comment (khiga8) [reply queries.sh]: Done in https://github.com/github/accessibility-alt-text-bot/pull/17/commits/ad9af45d3a480f82390b283cf5965af571d80548! :)

File: queries.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+# Given a discussion node ID, a message, and an optional reply to node ID, adds a discussion comment.
+function addDiscussionComment() {
+  local DISCUSSION_NODE_ID=$1
+  local MESSAGE=$2
+  local REPLY_TO_ID=$3
+
+  if [ -n "$REPLY_TO_ID" ]; then
+    gh api graphql -F discussionId="$DISCUSSION_NODE_ID" -F replyToId="$REPLY_TO_ID" -F body="$MESSAGE" -f query='
```
Comment (kendallgassner) [inline queries.sh:26]: Sorry! I meant could you do something like this:

```bash
if [ -n $REPLY_TO_ID ]; then; else REPLY="$REPLY_TO_ID"; fi
gh api graphql -F discussionId="$DISCUSSION_NODE_ID" -F replyToId=$REPLY -F body="$MESSAGE" -f query='
      mutation($discussionId: ID!, , $replyToId: ID, $body: String!) {
          addDiscussionComment(input: {discussionId: $discussionId, replyToId: $replyToId, body: $body}) {
          comment {
              id
          }
        }
      }
    '
```

^ I think the var can be passed in as `null` on graphql

Comment (kendallgassner) [reply queries.sh:26]: note: the if statement only is used to add `""` around `REPLY_TO_ID` if it is not null.

Comment (khiga8) [reply queries.sh:26]: Oh I see. I think I tried this before and ran into an error. Let me try again to confirm.

Comment (khiga8) [reply queries.sh:26]: We run into the following error, so I think we need to stick with two separate calls.

```bash
gh: Could not resolve to a node with the global id of ''
{"data":{"addDiscussionComment":null},"errors":[{"type":"NOT_FOUND","path":["addDiscussionComment"],"locations":[{"line":3,"column":9}],"message":"Could not resolve to a node with the global id of ''"}]}
```

Comment (kendallgassner) [reply queries.sh:26]: Oh weird it auto resolves it to an empty string.

File: queries.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      mutation($discussionId: ID!, , $replyToId: ID, $body: String!) {
```
Comment (kendallgassner) [inline queries.sh:27]:
```suggestion
      mutation($discussionId: ID!, $replyToId: ID, $body: String!) {
```

**Review by kendallgassner — APPROVED** [review]
❤️

**Review by Caprice2201 — COMMENTED** [review]
Submit
